src/view/screens/PostThread.tsx

```diff
@@ -23,62 +23,49 @@ type Props = NativeStackScreenProps<CommonNavigatorParams, 'PostThread'>
 export const PostThreadScreen = withAuthRequired(
   observer(function PostThreadScreenImpl({route}: Props) {
     const store = useStores()
+    const queryClient = useQueryClient()
     const minimalShellMode = useMinimalShellMode()
     const setMinimalShellMode = useSetMinimalShellMode()
     const safeAreaInsets = useSafeAreaInsets()
     const {name, rkey} = route.params
     const uri = makeRecordUri(name, 'app.bsky.feed.post', rkey)
-    const view = useMemo<PostThreadModel>(
-      () => new PostThreadModel(store, {uri}),
-      [store, uri],
-    )
     const {isMobile} = useWebMediaQueries()
 
     useFocusEffect(
       React.useCallback(() => {
         setMinimalShellMode(false)
-        const threadCleanup = view.registerListeners()
-
-        InteractionManager.runAfterInteractions(() => {
-          if (!view.hasLoaded && !view.isLoading) {
-            view.setup().catch(err => {
-              logger.error('Failed to fetch thread', {error: err})
-            })
-          }
-        })
-
-        return () => {
-          threadCleanup()
-        }
-      }, [view, setMinimalShellMode]),
+      }, [setMinimalShellMode]),
     )
 
     const onPressReply = React.useCallback(() => {
-      if (!view.thread) {
+      const post = getCachedPost(queryClient, uri)
+      if (!post) {
         return
       }
       store.shell.openComposer({
         replyTo: {
-          uri: view.thread.post.uri,
-          cid: view.thread.post.cid,
-          text: view.thread.postRecord?.text as string,
+          uri: post.uri,
+          cid: post.cid,
+          text:
+            'text' in post.record && typeof post.record.text === 'string'
+              ? post.record.text
+              : '',
           author: {
-            handle: view.thread.post.author.handle,
-            displayName: view.thread.post.author.displayName,
-            avatar: view.thread.post.author.avatar,
+            handle: post.author.handle,
+            displayName: post.author.displayName,
+            avatar: post.author.avatar,
           },
         },
-        onPost: () => view.refresh(),
+        // onPost: () => view.refresh(), TODO need to trigger this elsewhere
```

Comment:
We can probably create a helper function that takes in a RQ key and refreshes it, right?

Reply:
Yep, done